lib/src/main/java/co/nimblehq/alpine/lib/model/nfc/DocumentType.kt

```diff
@@ -0,0 +1,20 @@
+package co.nimblehq.alpine.lib.model.nfc
+
+private const val DOCUMENT_CODE_PASSPORT = "P"
+private const val DOCUMENT_CODE_ID_CARD = "I"
+
+enum class DocumentType {
+    PASSPORT,
+    ID_CARD,
+    OTHER;
+
+    companion object {
+        fun from(documentCode: String): DocumentType {
+            return when (documentCode) {
+                DOCUMENT_CODE_PASSPORT -> PASSPORT
+                DOCUMENT_CODE_ID_CARD -> ID_CARD
+                else -> OTHER
+            }
+        }
+    }
```

Comment:
Why don't we do this instead:
```suggestion
enum class DocumentType(val documentCode: String) {
    PASSPORT("P"),
    ID_CARD("I"),
    OTHER("");

    companion object {
            fun from(documentCode: String): DocumentType {
                return values().find { it.documentCode.equals(documentCode, true) } ?: OTHER
            }
        }
    }
```

Reply:
@lydiasama do you think it'll make more sense to expose `documentCode` to the consumer of the library? Also, what about `OTHER`? The consumer will get `""` if they call `OTHER.documentCode`. Not too meaningful, IMO. What's your thought on this?

Reply:
@AVI5HEK If the usage is the consumer will call this `from` function, so how about using this way?
```
enum class DocumentType {
    @Serializedname("P")
    PASSPORT,
    @Serializedname("I")
    ID_CARD,
    OTHER
```
So it will map to the `DocumentType` without `from` function and `documentCode`. Does it make more sense to you?

Reply:
@lydiasama now I'm lost. 😵‍💫 
![image](https://user-images.githubusercontent.com/8093908/179722593-c56702ed-d620-4e23-820c-cfcd99776de4.png)
I think we should make the companion object or the `from` method internal because it's meant to be used only internally. It should only be called from `NfcReaderImpl` unless the API consumer has reasons to call it. 🤔 Any thoughts?

Reply:
@AVI5HEK My bad. The `@Serializedname` is from Gson and we should not include it to our lib. How about `OTHER("Other")`? At least the consumer can see the `documentCode` as `Other` instead of the empty string.

Reply:
@lydiasama why should the consumer _need_ `documentCode`? 🤔

Reply:
@AVI5HEK The consumer doesn't need it. But as it is an enum, so it's okay to have it inside of the enum, isn't it? Otherwise, we will use the enum to be like another constant like a sealed class. 😅

Reply:
@AVI5HEK For that `documentCode` I don't think it's crucial when the consumer can see it when getting the `DocumentType`. Because after checking, we just use this `DocumentType` in `when` block to check its type. 😅

Reply:
@lydiasama my question is _why_ we should let them access that property? What exactly is the advantage if we use it or the disadvantage if we don't? 🤔

Reply:
@lydiasama went with your solution but made the `documentCode` property `private`. Please check. 🙇🏽

Reply:
@AVI5HEK Great! That's it!. 🎉